logging/google/cloud/logging/handlers/transports/background_thread.py

```diff
@@ -152,9 +151,10 @@ class BackgroundThreadTransport(Transport):
     def __init__(self, client, name):
         http = copy.deepcopy(client.connection.http)
         http = client.connection.credentials.authorize(http)
-        self.client = Client(client.project,
-                             client.connection.credentials,
-                             http)
+        # __class__ avoids a circular import
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2160
```

Comment:
That's not why the circular dependency exists, the class hierarchy is the "why". The cycle is `client --> handlers (package) --> transpors (package) --> background_thread --> client`.

But that dependency hierarchy is not god-given.

We won't let it block this PR, but it's more involved than just "no easy way to copy credentials". The number of modules is unruly and their dependency graph is also unruly.